src/compact_live.rs

```diff
@@ -0,0 +1,454 @@
+    /// - `noisy_uids`  : UIDs to compact (contains noise)
+    /// - `noise`       : Entry Table UIDs used as noise
+    async fn live_compact_uids(
+        &mut self,
+        rng: &mut impl CryptoRngCore,
+        k_value: &DemScheme::Key,
+        noisy_uids: Vec<Uid<UID_LENGTH>>,
+        noise: &HashSet<Uid<UID_LENGTH>>,
+    ) -> Result<(), Error<CustomError>> {
+        // Fetch all index entries (noisy).
+        let mut noisy_encrypted_entry_table = self
+            .fetch_entry_table(&HashSet::from_iter(noisy_uids.into_iter()))
+            .await?;
+
+        // Fetch chain data (noisy).
+        let chains = self
+            .fetch_chain_data(
+                k_value,
+                &noisy_encrypted_entry_table,
+            )
+            .await?;
+
+        // Select remaining locations (noisy).
+        let noisy_locations: HashSet<Location> = chains.chain_values
+            .iter()
+            .flat_map(|(_, values)| values)
+            .filter_map(|value| value.get_location())
+            .cloned()
+            .collect();
+        let noisy_remaining_locations = self.filter_removed_locations(&noisy_locations)?;
+
+        // While there are some Entry Table elements to compact...
+        while !noisy_encrypted_entry_table.is_empty() {
+            // Compute the new Entry Table (noisy) and compact chains (not noisy).
+            let (noisy_entry_table, new_chains) = self.compact_chains(
+                rng,
+                k_value,
+                noise,
+                &noisy_remaining_locations,
+                &noisy_encrypted_entry_table,
+                &chains.chain_values,
+            )?;
+
+            // Insert all recompacted chains first.
+            self.insert_chain_table(&new_chains.iter().flat_map(|(_, v)| v.clone()).collect()).await?;
+
+            // Try upserting the new entry table.
+            let upsert_data = UpsertData::new(
+                &noisy_encrypted_entry_table,
+                noisy_entry_table.encrypt::<DEM_KEY_LENGTH, DemScheme>(k_value, rng)?,
+            );
+
+            // These are failures to upsert.
+            noisy_encrypted_entry_table = self.upsert_entry_table(&upsert_data).await?;
+
+            // Delete unused chains (at least one chain element per upsert try):
+            let mut chains_to_delete: HashSet<Uid<UID_LENGTH>> =
+                HashSet::with_capacity(upsert_data.len());
+            chains_to_delete.extend(
+                // - new chains corresponding to unsuccessful upserts
+                new_chains
+                    .iter()
+                    .filter_map(|(uid, chains)| {
+                        if noisy_encrypted_entry_table.contains_key(uid) {
+                            Some(chains.keys())
+                        } else {
+                            None
+                        }
+                    })
+                    .flatten()
+                    .cloned(),
```

Comment:
I think a for loop here will be clearer (and shorter)
```suggestion
            for (uid, chains) in new_chains {
                if noisy_encrypted_entry_table.contains_key(&uid) {
                    chains_to_delete.extend(chains.0.into_keys());
                }
            }
```

It requires to `pub` the HashMap inside `EncryptedTable` (to access `.0`) or to have a implementation to take ownership of the inner HashMap (we currently have only function to get a reference to the inner HashMap). It also avoid the `.cloned()` at the end. No allocation needed since the `new_chains` is only used once